lib/internal/fs.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const Buffer = require('buffer').Buffer;
+const Stream = require('stream').Stream;
+const fs = require('fs');
+
+function assertEncoding(encoding) {
+  if (encoding && !Buffer.isEncoding(encoding)) {
+    throw new Error('Unknown encoding: ' + encoding);
+  }
+}
+
+// Temporary hack for process.stdout and process.stderr when piped to files.
+class SyncWriteStream extends Stream {
+  constructor(fd, options) {
+    super();
+    options = options || {};
+    this.fd = fd;
+    this.writable = true;
+    this.readable = false;
+    this.autoClose = options.autoClose === undefined ? true : options.autoClose;
+  }
+
+  write(data, arg1, arg2) {
+    var encoding, cb;
+
+    // parse arguments
+    if (arg1) {
+      if (typeof arg1 === 'string') {
+        encoding = arg1;
+        cb = arg2;
+      } else if (typeof arg1 === 'function') {
+        cb = arg1;
+      } else {
+        throw new Error('Bad arguments');
+      }
+    }
+    assertEncoding(encoding);
+
+    // Change strings to buffers. SLOW
+    if (typeof data === 'string') {
+      data = Buffer.from(data, encoding);
+    }
+
+    fs.writeSync(this.fd, data, 0, data.length);
+
+    if (cb) {
+      process.nextTick(cb);
+    }
+
+    return true;
+  }
+
+  end(data, arg1, arg2) {
+    if (data) {
+      this.write(data, arg1, arg2);
+    }
+    this.destroy();
+  }
+
+  destroy() {
+    if (this.autoClose)
+      fs.closeSync(this.fd);
+    this.fd = null;
+    this.emit('close');
+    return true;
+  }
+}
+
+// Alias destroySoon -> destroy
+SyncWriteStream.prototype.destroySoon = SyncWriteStream.prototype.destroy;
```

Comment:
do a quick search for `destroySoon` in `lib/*.js` and you'll see that there are several hits. If you'd like to pull it out, it should be done as a separate PR